File: java/src/main/java/io/openbanking/Connection.java
package io.openbanking;

/** A bank connection (consent). */
public record Connection(
        String sessionId,
        String aspspName,
        String aspspCountry,
        String validUntil,
        String status,
        long accountCount,
        String lastSyncedAt,
        String psuType) {
}

File: php/src/Model/Connection.php
<?php

declare(strict_types=1);

namespace OpenBankingIO\Model;

/** A bank connection (consent). */
final class Connection
{
    public function __construct(
        public readonly string $sessionId,
        public readonly string $aspspName,
        public readonly string $aspspCountry,
        public readonly ?string $validUntil,
        public readonly string $status,
        public readonly int $accountCount,
        public readonly ?string $lastSyncedAt,
        public readonly ?string $psuType,
    ) {
    }
}
